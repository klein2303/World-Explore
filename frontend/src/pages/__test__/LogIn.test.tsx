import {render, screen, fireEvent} from "@testing-library/react";
import {BrowserRouter} from "react-router-dom";
import {describe, it, expect} from "vitest";
import Login from "../LogIn"; // Assuming this is the correct path to your Login component

describe("Login Page", () => {
    const renderWithRouter = (ui: React.ReactElement) => {
        return render(ui, {wrapper: BrowserRouter});
    };

    it("renders the login page correctly", () => {
        renderWithRouter(<Login />);

        // Check that the logo and links are rendered
        expect(screen.getByLabelText("World explore")).toBeInTheDocument();
        expect(screen.getByLabelText("link to home page")).toBeInTheDocument();
        expect(screen.getByLabelText("Already registered?")).toBeInTheDocument();
        expect(screen.getByLabelText("Already registered")).toBeInTheDocument();

        // Check that the login form from LoginOrRegister component is rendered
        expect(screen.getByPlaceholderText("Email")).toBeInTheDocument();
        expect(screen.getByPlaceholderText("Password")).toBeInTheDocument();
        expect(screen.getByLabelText("Login button")).toBeInTheDocument();

        // Check that the image is rendered
        const image = screen.getByAltText("A beautiful landscape");
        expect(image).toBeInTheDocument();
        expect(image).toHaveAttribute("src", expect.stringContaining("loginImagePurple.jpg"));
    });

    it("handles login input changes", () => {
        renderWithRouter(<Login />);

        const emailInput = screen.getByPlaceholderText("Email");
        const passwordInput = screen.getByPlaceholderText("Password");

        // Simulate typing into the email and password fields
        fireEvent.change(emailInput, {target: {value: "testuser@example.com"}});
        fireEvent.change(passwordInput, {target: {value: "password123"}});

        // Check that the values are updated
        expect(emailInput).toHaveValue("testuser@example.com");
        expect(passwordInput).toHaveValue("password123");
    });

    it("submits the login form with valid credentials", () => {
        // Mock localStorage to simulate existing users
        localStorage.setItem(
            "userProfiles",
            JSON.stringify([{email: "testuser@example.com", password: "password123"}]),
        );

        renderWithRouter(<Login />);

        const emailInput = screen.getByPlaceholderText("Email");
        const passwordInput = screen.getByPlaceholderText("Password");
        const loginButton = screen.getByLabelText("Login button");

        // Simulate typing valid login credentials
        fireEvent.change(emailInput, {target: {value: "testuser@example.com"}});
        fireEvent.change(passwordInput, {target: {value: "password123"}});

        // Simulate form submission
        fireEvent.click(loginButton);

        // Check the feedback message for successful login
        expect(screen.getByLabelText("Log in successful")).toBeInTheDocument();
    });

    it("shows error when submitting login with invalid credentials", () => {
        renderWithRouter(<Login />);

        const emailInput = screen.getByPlaceholderText("Email");
        const passwordInput = screen.getByPlaceholderText("Password");
        const loginButton = screen.getByLabelText("Login button");

        // Simulate typing invalid login credentials
        fireEvent.change(emailInput, {target: {value: "wronguser@example.com"}});
        fireEvent.change(passwordInput, {target: {value: "wrongpassword"}});

        // Simulate form submission
        fireEvent.click(loginButton);

        // Check the feedback message for invalid login
        expect(screen.getByLabelText("No such email")).toBeInTheDocument();
    });
});
