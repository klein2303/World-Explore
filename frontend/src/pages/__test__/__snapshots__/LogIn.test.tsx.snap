// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Login Component > renders the Login component correctly with mock data 1`] = `
<DocumentFragment>
  <main
    aria-label="Login page"
    class="_registerPage_8c2a65"
  >
    <section
      aria-label="A section for inputs"
      class="_inputSection_8c2a65"
    >
      <section
        aria-label="Header of login page"
        class="_headerRegisterPage_8c2a65"
      >
        <section
          aria-label="Logo"
          class="_logo_8c2a65"
        >
          <a
            aria-label="link to home page"
            class="_logoLink_8c2a65"
            href="/ExploreCountries"
          >
            <p
              aria-label="World explore"
            >
              WorldExplore
            </p>
          </a>
        </section>
        <section
          aria-label="A link to login"
          class="_linkToLogIn_8c2a65"
        >
          <a
            aria-label="link"
            class="_navlinkOver_8c2a65"
            href="/Register"
          >
            <p
              aria-label="Already registered?"
            >
               Create an account
            </p>
          </a>
        </section>
      </section>
      <section
        aria-label="Login og register component"
      >
        <section
          aria-label="Login"
          class="_loginOrRegister_d1f9f8"
        >
          <section
            aria-label="Title"
            class="_title_d1f9f8"
          >
            <h3
              aria-label="Login page"
            >
              Log in
            </h3>
            <h4
              aria-label="Enter account details"
            >
              Enter your WorldExplore account details
            </h4>
          </section>
          <hr
            class="_line_d1f9f8"
          />
          <section
            aria-label="Input fields"
            class="_inputSections_d1f9f8"
          >
            <input
              aria-label="Email input field for login"
              class="_inputField_d1f9f8"
              name="email"
              placeholder="Email"
              type="email"
              value=""
            />
            <input
              aria-label="Password input field for login"
              class="_inputField_d1f9f8"
              name="password"
              placeholder="Password"
              type="password"
              value=""
            />
            <h5
              aria-label=""
            >
               
            </h5>
            <button
              aria-label="Login button"
              class="_submitButton_d1f9f8"
            >
              Log in
            </button>
          </section>
        </section>
      </section>
      <section
        aria-label="A link to login"
        class="_linkToLogInUnder_8c2a65"
      >
        <a
          aria-label="link"
          class="_navlinkUnder_8c2a65"
          href="/Register"
        >
          <p
            aria-label="Already registered"
          >
            Create an account
          </p>
        </a>
      </section>
    </section>
    <section
      aria-label="image"
      class="_image_8c2a65"
    >
      <img
        alt="A beautiful landscape"
        aria-label="A beautiful image of a landscape"
        class="_imageimage_8c2a65"
        height="700"
        src="/project2/loginImagePurple.jpg"
        width="700"
      />
    </section>
  </main>
</DocumentFragment>
`;
